lib/bazel/commands/target_list.go

```diff
@@ -0,0 +1,191 @@
+package commands
+
+import (
+	"io"
+	"regexp"
+	"sort"
+	"strings"
+
+	"golang.org/x/exp/maps"
+
+	bespb "github.com/enfabrica/enkit/third_party/bazel/buildeventstream"
+)
+
+// Assume that tests end in `_test`
+// BUG: DV regression tests also add a number suffix, like `test_12`.
+var labelIsTestRegexp = regexp.MustCompile(`_test(_\d+)?$`)
+
+func validStatuses() []string {
+	valid := maps.Keys(bespb.TestStatus_value)
+	for i := range valid {
+		valid[i] = strings.ToLower(valid[i])
+	}
+	return valid
+}
+
+type targetStatus struct {
+	targets  map[string]bespb.TestStatus
+	finished bool
+}
+
+func emptyTargetStatus() *targetStatus {
+	return &targetStatus{
+		targets:  map[string]bespb.TestStatus{},
+		finished: false,
+	}
+}
+
+func targetStatusFromBuildEvents(events []*bespb.BuildEvent) (*targetStatus, error) {
+	new := emptyTargetStatus()
+
+	for _, event := range events {
+		switch id := event.Id.Id.(type) {
+		case *bespb.BuildEventId_TargetConfigured:
+			new.targets[id.TargetConfigured.GetLabel()] = bespb.TestStatus_NO_STATUS
+
+		case *bespb.BuildEventId_TargetCompleted:
+			lbl := id.TargetCompleted.GetLabel()
+			if !labelIsTestRegexp.MatchString(lbl) {
+				new.targets[lbl] = bespb.TestStatus_PASSED
```

Comment:
Is this meant to be hard coded to PASSED? Double checking to make sure Completed means passed and not completed/success || completed/failed type of thing

Reply:
Yeah - some targets are build only, while some are tests. Here I'm trying to check if a target is build-only, and if so, then the TargetCompleted message holds the final status - if it's a test, we're waiting for a TestResult message for the final status (it should be considered as "still running")

I'll add a comment here to this effect - also the way tests were detected is error-prone, so I beefed this portion up a bit.